test/integration/apiserver/crd_validation_expressions_test.go

```diff
@@ -888,6 +957,76 @@ var structuralSchemaWithBlockingErr = []byte(`
   }
 }`)
 
+var structuralSchemaWithItemsUnderArray = []byte(`
+{
+  "openAPIV3Schema": {
+    "description": "CRD with CEL validators",
+    "type": "object",
+    "properties": {
+      "spec": {
+        "type": "object",
+        "properties": {
+          "backend": {
+            "type": "array",
+            "maxItems": 100,
+            "items": {
+              "type": "object",
+              "properties": {
+                "replicas": {
+                  "type": "integer"
+                }
+              },
+              "required": [
+                "replicas"
+              ],
+              "x-kubernetes-validations": [
+                {
+                  "rule": "0 <= self.replicas && self.replicas <= 10"
+                }
+              ]
+            }
+          }
+        }
+      }
+    }
+  }
+}`)
+
+var structuralSchemaWithItemsUnderObject = []byte(`
```

Comment:
How about properties under array and similar cases? (Doesn't necessarily need to be an integration tests for all of these)

Reply:
Added test in schema.test and fixed the previous bug for nil check. Thanks